src/riot-rs-embassy/src/arch/dummy/executor.rs

```diff
@@ -24,9 +26,15 @@ impl Executor {
 pub struct Spawner;
 
 impl Spawner {
-    #[allow(clippy::result_unit_err)]
+    #[allow(
+        clippy::result_unit_err,
+        clippy::needless_pass_by_value,
```

Comment:
This is also Clippy subtly telling us that these should really be implementing a trait, even if only one type implementing the trait is ever available on a system. (But disabling the lints is a reasonable step at this point).